lib/src/signed_video_h26x_auth.c

```diff
@@ -128,9 +146,31 @@ verify_gop_hash(signed_video_t *self)
   return (memcmp(gop_info->computed_gop_hash, self->received_gop_hash, hash_size) == 0);
 }
 
-/**
- * Adds the hashes of NALUs used in computing the GOP hash and marks these NALUs as part of the GOP
- * hash.
+/*
+ * Iterates through the NALU list to find the first NALU used in the GOP hash.
+ * If the linked hash has not yet been updated with this NALU's hash, it updates the
+ * linked hash with the first NALU hash and marks it as used.
+ */
+void
+update_link_hash_for_auth(signed_video_t *self, h26x_nalu_list_t *nalu_list)
```

Comment:
You don't need `nalu_list` as input, since you feed this function with `self->nalu_list`.